bananabot.py

```diff
@@ -26,80 +26,115 @@ def verify_password(username, password):
-}
+banana_time_announcement = Announcement(datetime.time(15, 30, 0), "# @HERE Banana Time!", datetime.time(15, 30, 0))
+banana_time_announcement.id = "banana_time"
 
-minsBeforeAnnouncements = {
-    60: MinsBeforeAnnouncement(60, "60 Minutes until Banana Time!"),
-    30: MinsBeforeAnnouncement(30, "30 Minutes until Banana Time!"),
-    10: MinsBeforeAnnouncement(10, "10 Minutes until Banana Time!")
-}
+
+# Initial announcements
+initial_announcements = [
+    banana_time_announcement,
+    Announcement(datetime.time(10, 0, 0), "Banana time is at 15:30 today!", datetime.time(15, 30, 0)),
+    MinsBeforeAnnouncement(60, "Banana time is in 60 minutes!", datetime.time(15, 30, 0)),
+    MinsBeforeAnnouncement(30, "Banana time is in 30 minutes!", datetime.time(15, 30, 0)),
+    MinsBeforeAnnouncement(10, "Banana time is in 10 minutes!", datetime.time(15, 30, 0))
+]
+
+for announcement in initial_announcements:
+    announcements[announcement.id] = announcement
 
 
 # Banana Time Functions
+def get_banana_time():
+    return announcements['banana_time'].time
+
 def start():
-    for announcement in timeAnnouncements.values():
-        worker = TimeAnnouncementWorker(announcement.time, announcement.text)
-        worker.start()
-        workers.append(worker)
-    
-    for announcement in minsBeforeAnnouncements.values():
-        worker = MinsBeforeAnnouncementWorker(announcement.minsBefore, announcement.text, timeAnnouncements['bananaTime'].time)
-        worker.start()
-        workers.append(worker)
+    for key in announcements:
+        if isinstance(announcements[key], MinsBeforeAnnouncement):
+            worker = MinsBeforeAnnouncementWorker(announcements[key].id, announcements[key].mins_before, announcements[key].text, announcements[key].banana_time)
+            workers[worker.id] = worker
+            workers[worker.id].start()
+        else:
+            worker = AnnouncementWorker(announcements[key].id, announcements[key].time, announcements[key].text, announcements[key].banana_time)
+            workers[worker.id] = worker
+            workers[worker.id].start()
+
 
 def stop():
-    for worker in workers:
-        worker.terminate()
+    for key in workers:
+        workers[key].terminate()
     
-    globals()['workers'] = []
+    globals()['workers'] = {}
 
 def update():
     if active:
         stop()
         start()
 
-def stringToTime(newTime):
-    t = time.strptime(newTime, "%H:%M")
+def string_to_time(new_time):
+    t = time.strptime(new_time, "%H:%M")
     return datetime.time(hour = t.tm_hour, minute = t.tm_min)
 
-def setBananaTime(time):
+def set_banana_time(time):
     app.logger.debug("Requested banana time " + time)
-    timeAnnouncements['bananaTime'].time = stringToTime(time)
-    app.logger.info("New banana time set at " + str(timeAnnouncements['bananaTime'].time))
+    time = string_to_time(time)
+    announcements['banana_time'].time = time
```

Comment:
What is the difference between `.time` and `.banana_time` in this context?

Reply:
`.time` is a reference to what time the `AnnouncementWorker` will send a request. `.banana_time` is a reference to Banana Time.  This is changing though.

Per our discussions, `banana_time` will now be a class variable of the `Announcement` class. That should solve this weird case. I think for now it is probably okay to leave the `AnnouncementWorker` taking `banana_time` as a parameter. The workers `sleep()` so even if it was a class variable it wouldn't change anything, and the logic is likely to change with the introduction of a manager so for now it is probably okay for it to remain as is.